cpp/test/ApprovalTests.hpp

```diff
@@ -0,0 +1 @@
+#include "ApprovalTests.v.7.0.0.hpp"
```

Comment:
I wonder whether maintenance would be a bit easier if this file was in the include directory, alongside the file it is including - so that if the ApprovalTests version is changed, it's a bit more obvious to change this one too...

Alternatively, the CMake download rule could be changed to remove the version from the name of the downloaded file....

Reply:
It become a bit of compromise between the non-working meson wrapper and CMake. I had them both in include, then removed the include folder when I got CMake to download nicely. Then when I didn't like to spend more time on the meson build I readded the include folder but had already moved the version wrapper into tests.

Reply:
That makes sense... There was a point when I was writing comments that I wondered whether to suggest dropping the meson support - but then I thought you would only have added it if you needed it...

Reply:
I started with meson as it simpler and cleaner syntax, but then while working with CLion I learned that CLion doesn't like to work without CMake. Then I noticed that there have been a few nice additions to later CMake so it not as hard to work and download external projects as it used to be. So I made the most of the work with CMake and just, in the end, made sure meson still worked. I still haven't tried this code anyplace but my MacBook.